src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Syntax/SyntaxNode.cs

```diff
@@ -601,8 +601,17 @@ SyntaxToken walkForward(PooledArrayBuilder<(SyntaxNode node, int nodeIndexInPare
                     }
                 }
 
-                // Got to the end of the node without finding a desired token. Pop up the stack and try again
                 foundToken = null;
+
+                // The start of the document is the only special case:
+                //  - If we're walking backwards and hit the start of the document, we need to treat this as if we should walk forward. There's no
+                //    previous node to attach the token to, so walking forward is the only option.
+                if (walkBackwards && parent!.SpanStart == 0)
```

Comment:
Seems like `!` shouldn't be needed (not only here, but also above in this method).

```suggestion
                if (walkBackwards && parent.SpanStart == 0)
```

Reply:
`Debug.Assert` is not annotated in .netstandard 2.0, so unfortunately it is necessary.